parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -405,21 +406,26 @@ def worker_watchdog(self, kill_event: threading.Event):
                                                             self.pending_result_queue,
                                                             self.ready_worker_queue,
                                                             self._tasks_in_progress,
-                                                            self.cpu_affinity),
+                                                            self.cpu_affinity,
+                                                            self.heartbeat_period),
                                        name="HTEX-Worker-{}".format(worker_id))
                     self.procs[worker_id] = p
                     logger.info("Worker {} has been restarted".format(worker_id))
 
         logger.critical("Exiting")
 
+    def _handle_sigterm(self, signum, frame):
+        logger.critical("Received SIGTERM; setting kill event")
+        self._kill_event.set()
+
```

Comment:
for historical context, worker pool STOP was introduced to `mpix`, an earlier iteration of the high throughput executor: 1c3d1a352de26b70ecb50a245ff6b2b08341d76f

Reply:
We also set the `kill_event` [here](https://github.com/Parsl/parsl/blob/741293a76c2ad114b48bcc927701258b7ba7f355/parsl/executors/high_throughput/process_worker_pool.py#L319) when the manager loses contact with the interchange, but your point stands.

Reply:
Och; sorry I'm late to this party; until educated otherwise, I've fairly strictly followed the mantra "do as little as possible in a signal handler."  That may be outmoded thinking given that we're in Python's interpreter at this point, but it has served me quite well when in C-land.

I bring it up as a potential issue because I observed (and fixed) an issue last year in funcX-land that, as far as I could tell, was a deadlock while attempting to set an event while in a signal handler.

Meaning, I suggest instead to do the smallest thing you can do in this signal handler, and follow up elsewhere ASAP.  I've found "set a boolean" that is then checked at appropriate times elsewhere to be a winning strategy.

Reply:
i think there's probably a potential hang in that log statement if the thread that was interrupted to run the signal handler happens to have the locking log (a variant of the classic "why we can't use fork" hang)

Reply:
You both bring up good points. I've simplified the signal handlers to only set a `self._time_to_stop` boolean flag, which the main thread will check every 0.1 seconds. When the flag is set, we will, in turn, set `self._kill_event`. 

The long-term solution is likely to replace the threading event entirely with the flag, but that will require a larger refactor than I intend for this PR. For example, we might trigger the worker watchdog loop when the manager receives a SIGCHLD rather than every heartbeat period.